package/MDAnalysis/coordinates/chain.py

```diff
@@ -371,17 +391,11 @@ def __iter__(self):
         for ts in self.__chained_trajectories_iter:
             yield ts
 
-    @staticmethod
```

Comment:
I think you should keep this because it deals with the case where you can create a chained trajectory with different formats:
```python
u = Universe(topology, [("file1.cor", "CRD"), ("file2.trr", "TRR"), ("file3.ent", "PDB")])
```

Reply:
Oh is this option documented somewhere? But even so I would make it a separate method and not a static method.

Reply:
Not really... [our Universe docs](https://www.mdanalysis.org/mdanalysis/documentation_pages/core/universe.html) do not even document the `trajectory` argument properly :-(. 

But the following works (see #76):
```python
>>> import MDAnalysis as mda
>>> from MDAnalysis.tests.datafiles import GRO, PDB, XTC, TRR, TPR
>>> u = mda.Universe(TPR, [(GRO, "gro"), (PDB, "pdb")])
>>> u.trajectory.n_frames
2
```
and it is tested in TestChainReaderFormats https://github.com/MDAnalysis/mdanalysis/blob/74d51a22be9fe8738313c2e78f72eb0d19b8a792/testsuite/MDAnalysisTests/coordinates/test_chainreader.py#L142

Reply:
I had made it a static method because it does not need self. When we had the quality control bot (QuantifiedCode?) it used to suggest to make anything static that does not use self. Makes some sense to me. 

Why do you prefer it to me a normal method?

Reply:
Sorry I meant function. There is no need for it in the class.

Reply:
Yes, at the time I wanted to avoid cluttering the module space. If you find a better place for it, please, feel free to move it elsewhere. Perhaps `lib.util`?

Reply:
I now do the filenames parsing in `__init__` and remove the eventual tuple that is passed. It works and we don't need to keep this function around.